oddt/surface.py

```diff
@@ -0,0 +1,128 @@
+    method from scikit-image.
+
+    Parameters
+    ----------
+    molecule : oddt.toolkit.Molecule object
+        Molecule for which the surface will be generated
+
+    remove_hoh : bool (default = False)
+        If True, remove waters from the molecule before generating the surface
+
+    scaling : float (default = 1)
+        Expands the grid in which computation is done by a factor of scaling.
+        Results in a more accurate representation of the surface, but increases
+        computation time.
+
+    probe_radius : float (default = 1.4)
+        Radius of a ball used to patch up holes inside the molecule
+        resulting from some molecular distances being larger
+        (usually in protein). Basically reduces the surface to one
+        accesible by other molecules of radius smaller than probe_radius.
+
+    Returns
+    -------
+    verts : numpy array
+        Spatial coordinates for mesh vertices.
+
+    faces : numpy array
+        Faces are defined by referencing vertices from verts.
+    """
+    atom_dict = molecule.atom_dict
+    atom_dict = atom_dict[atom_dict['atomicnum'] != 1]
+    if remove_hoh:
+        assert molecule.protein is True, ('Residue names are needed for water removal,'
+                                          'molecule.protein property must be set to True')
+        no_hoh = atom_dict['resname'] != 'HOH'
+        atom_dict = atom_dict[no_hoh]
+
+    # Take a molecule's coordinates and atom radii and scale if necessary
+    coords = atom_dict['coords'] * scaling
+    radii = atom_dict['radius'] * scaling
+
+    # Create a ball for each atom in the molecule
+    ball_dict = {radius: ball(radius, dtype=bool) for radius in set(radii)}
+    ball_radii = np.array([ball_dict[radius].shape[0] for radius in radii])
+
+    # Transform the coordinates because the grid starts at (0, 0 ,0)
+    min_coords = np.min(coords, axis=0)
+    max_rad = np.max(ball_radii, axis=0)
+    adjusted = np.round(coords - min_coords + max_rad * 5).astype(np.int64)
+    offset = adjusted[0] - coords[0]
+
+    # Calculate boundries in the grid for each ball.
+    ball_coord_min = (adjusted.T - np.floor(ball_radii / 2).astype(np.int64)).T
+    ball_coord_max = (ball_coord_min.T + ball_radii).T
+
+    # Create the grid
+    grid = np.zeros(shape=ball_coord_max.max(axis=0) + int(8 * scaling), dtype=bool)
+    # Place balls in grid
+    for radius, coord_min, coord_max in zip(radii, ball_coord_min, ball_coord_max):
+        grid[coord_min[0]:coord_max[0],
+             coord_min[1]:coord_max[1],
+             coord_min[2]:coord_max[2]] += ball_dict[radius]
+    spacing = (1 / scaling,) * 3
+
+    # Hole-filling with morphological closing
+    grid = binary_closing(grid, ball(probe_radius * 2 * scaling))
+
+    # Marching cubes
+    verts, faces, _, _ = measure.marching_cubes_lewiner(grid, level=0, step_size=1, spacing=spacing)
+    return verts - offset / scaling, faces
```

Comment:
I asked you to add a comment here in the previous review: [discussion_r133442155](http://github.com/oddt/oddt/pull/37#discussion_r133442155)